src/components/KAlert/KAlert.vue

```diff
@@ -1,578 +1,185 @@
-      <slot
-        v-if="hasActionButtons"
-        name="actionButtons"
-      >
-        <KButton
-          size="small"
-          @click="proceed"
-          @keyup.enter="proceed"
-        />
-      </slot>
-
-      <KButton
-        v-if="dismissType === 'button'"
-        size="small"
-        @click="dismissAlert"
-      >
-        Dismiss
-      </KButton>
-    </div>
+    <CloseIcon
+      v-if="dismissible"
+      class="alert-dismiss-icon"
+      :size="KUI_ICON_SIZE_40"
+      tabindex="0"
+      @click="$emit('dismiss')"
+      @keydown.space.prevent
+      @keyup.enter="$emit('dismiss')"
+      @keyup.space="$emit('dismiss')"
+    />
   </div>
 </template>
 
 <script setup lang="ts">
+import { computed } from 'vue'
 import type { PropType } from 'vue'
-import { computed, useSlots } from 'vue'
-import KButton from '@/components/KButton/KButton.vue'
-import KIcon from '@/components/KIcon/KIcon.vue'
-import type { AlertAppearance, AlertDismissType, AlertSize, AlertType } from '@/types'
+import type { AlertAppearance } from '@/types'
 import { AlertAppearances } from '@/types'
-import { KUI_ICON_SIZE_20, KUI_ICON_SIZE_60 } from '@kong/design-tokens'
+import { InfoIcon, CheckCircleIcon, WarningIcon, ClearIcon, CloseIcon } from '@kong/icons'
+import { KUI_ICON_SIZE_40 } from '@kong/design-tokens'
 
-defineProps({
-  /**
-    * Message to show in alert
-    */
-  alertMessage: {
-    type: String,
-    default: '',
-  },
-  /**
-     * Set whether or not the alert box is shown.
-     */
-  isShowing: {
-    type: Boolean,
-    default: true,
-  },
-  /**
-     * Fixes alert to top
-     */
-  isFixed: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-     * Set whether or not alert has full border is visible
-     */
-  isBordered: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-     * Sets whether or not alert has left border
-     */
-  hasLeftBorder: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-     * Sets whether or not alert has right border
-     */
-  hasRightBorder: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-     * Sets whether or not alert has top border
-     */
-  hasTopBorder: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-     * Sets whether or not alert has bottom border
-     */
-  hasBottomBorder: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-     * Center text inside alert
-     */
-  isCentered: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-     * Set alert box icon size
-     */
-  iconSize: {
-    type: String,
-    default: KUI_ICON_SIZE_60,
-  },
-  /**
-     * Set alert box type of icon
-     */
-  icon: {
-    type: String,
-    default: '',
-  },
-  /**
-     * Set alert box icon color
-     */
-  iconColor: {
-    type: String,
-    default: '',
-  },
-  /**
-     * Alert message title
-     */
+type AlertIcon = typeof InfoIcon | typeof CheckCircleIcon | typeof WarningIcon | typeof ClearIcon
```

Comment:
nit: all of these icons actually have the same interface, so you could just type it as one of them if you wanted